### DISCUSSION TITLE
Logging HTTP headers

### DISCUSSION THREAD

**Discussion by jbakermk** [encode/httpx#1975] [General]
I've looked at the output from HTTPX_LOG_LEVEL=trace, and there's a lot of debugging. That's great. But if writing a user-facing application where a simplified set of logging of HTTP headers (on request and response) would be desirable for troubleshooting, trace logging is too much.

HTTPX_LOG_LEVEL=debug provides the method and URL. Can I suggest it's extended to neatly debug the headers,  i.e. a logging.debug(..) for each header, value? This would be really quick to add in any PR and would hugely benefit the user debugging experience from a CLI written around httpx.

**Comment by lovelydinosaur** [comment]
I'm not sure, but I'd be happy for us to review it.
What do we currently log at `INFO` level?

**Reply by jbakermk** [reply]
Nothing really!
